### PR TITLE
Updated twitter_classifier Reference App to Spark 2 and Scala 2.11

### PR DESCRIPTION
**Pull request by mslinn**
Summary of changes to the `twitter_classifier` reference app:

- Updated from Spark 1.4 and Scala 2.10 to Spark 2 and Scala 2.11
- Specified the SBT version as v0.13.13, previously no version of SBT had been specified
- Rewrote 3 command-line programs so they look like Scala programs instead of Java programs.
- Factored out common code for the 3 command-line programs
- Provided scripts to drive the 3 command-line programs
- Improved docs; still not quite happy with how the overall docs are laid out, might be good to discuss how to best bring it all together.

Although I am quite comfortable with Scala, this is my first time with Spark 2, so suggestions on how to best demonstrate the API would be most welcome.

### CODE REVIEW

File: twitter_classifier/scala/src/main/scala/com/databricks/apps/twitterClassifier/package.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+package twitterClassifier {
+  import org.apache.spark.SparkContext
+  import org.apache.spark.sql.SparkSession
+
+  object SparkSetup {
+    val spark = SparkSession
+      .builder
+      .appName(getClass.getSimpleName.replace("$", ""))
+      .master("local[*]")
```
Comment (tashoyan) [inline twitter_classifier/scala/src/main/scala/com/databricks/apps/twitterClassifier/package.scala]: Why master is hardcoded? Don't we rather expect user to specify his master URL in command line when launching the application?

**Comment by mslinn** [comment]
Thank you for the suggestion. I will remove master :)

Mike

**Comment by mslinn** [comment]
I wrote up a brief [blog post](https://medium.com/@mslinn/lessons-from-updating-the-twitter-classifier-apache-spark-reference-application-3c841bd69c0b#.mx9fgxidt) on this new version of the Twitter Classifier Reference Application.

**Comment by tashoyan** [comment]
@mslinn, your post inspired me to write Scala implementation of Log Analyzer Application.

* [Commander Scala](https://github.com/acrisci/commander-scala) to parse command-line arguments
* [Scala ARM](http://jsuereth.com/scala-arm/) to avoid system resource leaks
* Rich Scala collections API to prepare data for the report
* Scala Option to safely operate with probably absent objects

Scala is awesome.

**Comment by mslinn** [comment]
@tashoyan Excellent! I'd be happy to do an idiomatic Scala style review, when ready. Let me know!

**Comment by tashoyan** [comment]
@mslinn My pull request is ready: https://github.com/databricks/reference-apps/pull/89
This is my first experience with Scala, so I would be grateful to you for a review.

**Pull request merged**